crates/iceberg/src/writer/file_writer/rolling_writer.rs

```diff
@@ -15,67 +15,138 @@
 // specific language governing permissions and limitations
 // under the License.
 
+use std::fmt::{Debug, Formatter};
+
 use arrow_array::RecordBatch;
 
-use crate::spec::DataFileBuilder;
+use crate::io::{FileIO, OutputFile};
+use crate::spec::{DataFileBuilder, PartitionKey};
 use crate::writer::CurrentFileStatus;
+use crate::writer::file_writer::location_generator::{FileNameGenerator, LocationGenerator};
 use crate::writer::file_writer::{FileWriter, FileWriterBuilder};
 use crate::{Error, ErrorKind, Result};
+//
+// /// Builder for creating a `RollingFileWriter` that rolls over to a new file
+// /// when the data size exceeds a target threshold.
+// #[derive(Clone)]
+// pub struct RollingFileWriterBuilder<
```

Comment:
I think we will no longer need a builder for RollingFileWriter since it will be injected into IcebergWriter directly